tests/Integration/Behaviour/Features/Scenario/Feature/feature_management.feature

```diff
@@ -8,22 +8,121 @@ Feature: Product feature management
 
   Background:
     Given shop "shop1" with name "test_shop" exists
+    And single shop "shop1" context is loaded
+    And language "en" with locale "en-US" exists
+    And language with iso code "en" is the default one
+    And language "language2" with locale "fr-FR" exists
 
-  Scenario: Create product feature
+  Scenario: Create product feature in single shop
     When I create product feature "feature1" with specified properties:
-      | name | My feature |
-    Then product feature "feature1" name should be "My feature"
-
-  Scenario: Update product feature name
-    Given product feature with reference "feature1" exists
-    When I update product feature with reference "feature1" field "name" in default language to "My great feature"
-    Then product feature with reference "feature1" field "name" in default language should be "My great feature"
-
-  Scenario: Updating product feature with empty name should not be allowed
-    Given product feature with reference "feature1" exists
-    When I update product feature with reference "feature1" field "name" in default language to ""
-    Then I should get an error that feature name is invalid.
-
-  Scenario: Creating product feature with empty name should not be allowed
-    When I create product feature with empty name
-    Then I should get an error that feature name is invalid.
+      | name[en-US] | My feature en |
+      | name[fr-FR] | My feature fr |
+    Then product feature feature1 should have following details:
+      | name[en-US]      | My feature en |
+      | name[fr-FR]      | My feature fr |
+      | associated shops | shop1         |
+
+  Scenario: Update product feature
+    Given I create product feature "feature2" with specified properties:
+      | name[en-US] | My feature en |
+      | name[fr-FR] | My feature fr |
+    And product feature feature2 should have following details:
+      | name[en-US]      | My feature en |
+      | name[fr-FR]      | My feature fr |
+      | associated shops | shop1         |
+    When I update product feature feature2 with following details:
+      | name[en-US] | My feature en updated1 |
+      | name[fr-FR] | My feature fr updated1 |
+    Then product feature feature2 should have following details:
+      | name[en-US] | My feature en updated1 |
+      | name[fr-FR] | My feature fr updated1 |
+
+  Scenario: Creating and updating feature with empty shop association should assign the feature to current shop context by default
+    When I create product feature "feature3" with specified properties:
+      | name[en-US]      | My feature en |
+      | name[fr-FR]      | My feature fr |
+      | associated shops |               |
+    Then product feature feature3 should have following details:
+      | name[en-US]      | My feature en |
+      | name[fr-FR]      | My feature fr |
+      | associated shops | shop1         |
+    When I update product feature feature3 with following details:
```

Comment:
object model always assigns shops based on context (if you are in single shop context and leave it empty it will be assigned to that shop, or if you are in all shops context it will even be assinged to all shops)

Reply:
Yeah but I don't think we should count on the ObjectModel magic fallback in our command implementation, as much as possible we should be in control and make the modifications explicit. So if no shop IDs is defined in the command no changes should be applied regardless of the context

And since no association is kind of similar to a deletion, but already said that only the delete action deletes the entity, maybe the command shouldn't accept an empty array? Null is fine, no modif, but empty array could be considered invalid. What do you think?